src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -770,11 +786,26 @@ internal Binder[] GetAllBinders()
             patterns.Free();
         }
 
-        internal BoundExpression WrapWithVariablesIfAny(CSharpSyntaxNode node, BoundExpression expression)
+        internal BoundExpression WrapWithVariablesIfAny(CSharpSyntaxNode scopeDesignator, BoundExpression expression)
         {
-            return (Locals.Length == 0)
+            var locals = this.GetDeclaredLocalsForScope(scopeDesignator);
+            return (locals.Length == 0)
                 ? expression
-                : new BoundSequence(node, Locals, ImmutableArray<BoundExpression>.Empty, expression, expression.Type) { WasCompilerGenerated = true };
+                : new BoundSequence(scopeDesignator, locals, ImmutableArray<BoundExpression>.Empty, expression, expression.Type) { WasCompilerGenerated = true };
+        }
+
+        internal BoundStatement WrapWithVariablesIfAny(CSharpSyntaxNode scopeDesignator, BoundStatement statement)
+        {
+            var locals = this.GetDeclaredLocalsForScope(scopeDesignator);
+            if (locals.IsEmpty)
```

Comment:
the expression helper above uses   locals.Length == 0, perhaps this one should as well?